frontend/src/context/ws-client-provider.tsx

```diff
@@ -116,7 +126,11 @@ export function WsClientProvider({
       const baseUrl =
         import.meta.env.VITE_BACKEND_BASE_URL || window?.location.host;
       const protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
-      ws = new WebSocket(`${protocol}//${baseUrl}/ws`, [
+      let wsUrl = `${protocol}//${baseUrl}/ws`;
+      if (events.length) {
+        wsUrl += `?latest_event_id=${events[events.length - 1].id}`;
```

Comment:
This might fix https://github.com/All-Hands-AI/OpenHands/issues/4938 or a part of it. It's still very strange that  only some messages are loaded sometimes.